nixos/modules/services/desktops/dunst.nix

```diff
@@ -0,0 +1,538 @@
+                        and left of "1".
+                    '';
+                };
+
+                context = mkOption {
+                    type = types.str;
+                    default = "ctrl+shift+period";
+                    description = ''
+                        Context menu.
+                    '';
+                };
+
+            };
+
+            urgency = {
+                low = {
+                    background = mkOption {
+                        type = types.str;
+                        default = ''#222222'';
+                        description = ''
+                        '';
+                    };
+
+                    foreground = mkOption {
+                        type = types.str;
+                        default = ''#888888'';
+                        description = ''
+                        '';
+                    };
+
+                    timeout = mkOption {
+                        type = types.int;
+                        default = 10;
+                        description = ''
+                        '';
+                    };
+                };
+                normal = {
+                    background = mkOption {
+                        type = types.str;
+                        default = ''#285577'';
+                        description = ''
+                        '';
+                    };
+
+                    foreground = mkOption {
+                        type = types.str;
+                        default = ''#ffffff'';
+                        description = ''
+                        '';
+                    };
+
+                    timeout = mkOption {
+                        type = types.int;
+                        default = 10;
+                        description = ''
+                        '';
+                    };
+                };
+                critical = {
+                    background = mkOption {
+                        type = types.str;
+                        default = ''#900000'';
+                        description = ''
+                        '';
+                    };
+
+                    foreground = mkOption {
+                        type = types.str;
+                        default = ''#ffffff'';
+                        description = ''
+                        '';
+                    };
+
+                    timeout = mkOption {
+                        type = types.int;
+                        default = 0;
+                        description = ''
+                        '';
+                    };
+                };
+            };
+        };
+    };
+
+
+###### implementation
+
+    config = mkIf config.services.dunst.enable {
+
+        environment.systemPackages = [ pkgs.dunst pkgs.dmenu ];
+
+        systemd.services.dunst = {
+            description = "Dunst Daemon";
+            wantedBy    = [ "multi-user.target" ];
+
+            serviceConfig.ExecStart = "${pkgs.dunst}/bin/dunst -config ${dunstConf}";
```

Comment:
To me, it makes sense to pass the dmenu executable path here instead of adding dmenu to the system package set.